script/migrate-test-e2e.js

```diff
@@ -8,10 +9,43 @@ const emailGithub = "github@joshuakgoldberg.com";
 const emailNpm = "npm@joshuakgoldberg.com";
 const owner = "JoshuaKGoldberg";
 const title = "Create TypeScript App";
+import { createRootFiles } from "../lib/steps/writing/creation/rootFiles.js";
 
-await $({
-	stdio: "inherit",
-})`c8 -o ./coverage-migrate -r html -r lcov --src src node ./bin/index.js --base everything --mode migrate --description ${description} --email-github ${emailGithub} --email-npm ${emailNpm} --owner ${owner} --title ${title} --repository ${repository} --skip-all-contributors-api --skip-github-api --skip-install`;
+const args = {
+	"--base": ["base", "everything"],
+	"--description": ["description", description],
+	"--email-github": ["", emailGithub],
+	"--email-npm": ["", emailNpm],
+	"--mode": ["mode", "migrate"],
+	"--owner": ["owner", owner],
+	"--repository": ["repository", repository],
+	"--skip-all-contributors-api": ["skipAllContributorsApi", ""],
+	"--skip-github-api": ["skipGitHubApi", ""],
+	"--skip-install": ["skipInstall", ""],
+	"--title": ["title", title],
+};
+
+const argsStr = Object.entries(args)
+	.map(([k, v]) => `${k}${v[1] ? ` ${v[1]}` : ""}`)
+	.join(" ");
+const argsObj = Object.fromEntries(
+	Object.values(args)
+		.map(([k, v]) => [k, v !== "" ? v : true])
+		.filter(([k]) => !!k),
+);
+
+const rootFiles = await createRootFiles({
+	email: {
+		github: args["--email-github"][1],
+		github: args["--email-npm"][1],
+	},
+	...argsObj,
+});
```

Comment:
[Refactor] This is a nifty approach, basing the expected output files on the function that's meant to create them (`createRootFiles`). Unfortunately that sets up something of a circular dependency: that we're setting up the test for expected files on source, but that same source is essentially what we're trying to test! Which is why the bug in the test of `email` having two `github` keys (instead of `github` and `npm`) wasn't caught.

Unfortunately, I don't see a way around having this test file explicitly list out the expected changes. Hopefully they should be pretty minimal and not too many lines per file.